### PR TITLE
PUBDEV-7507 - Deprecate h2o-scala module.

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-7507

### CODE REVIEW

File: h2o-scala/README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
-# H<sub>2</sub>O Scala API
+# H<sub>2</sub>O Scala API (Deprecated)
 ----
+**Deprecation warning**: This module is now deprecated and will be released in next major version `3.32.0.1`.
```
Comment (jakubhava) [inline h2o-scala/README.md:3]: and will be released -> and will be removed

Comment (Pscheidl) [reply h2o-scala/README.md:3]: Thanks !

Comment (Pscheidl) [inline h2o-scala/README.md:3]:
```suggestion
**Deprecation warning**: This module is now deprecated and will be removed in next major version `3.32.0.1`.
```

File: h2o-docs/src/product/api-changes.rst [hunk 2]
```diff
@@ -3,6 +3,12 @@ API-Related Changes
 
 H2O-3 does its best to keep backwards compatibility between major versions, but sometimes breaking changes are needed in order to improve code quality and to address issues. This section provides a list of current breaking changes between specific releases.
 
+From 3.30.0.5
+~~~~~~~~~~~~~
+
+The ``h2o-scala`` module is deprecated in version 3.30.0.5 and will be completely removed in the next major version, 3.32.0.1. This by no means has any impact on Scala support, as the ``h2o-scala`` module was intended for internal use only and does not represent the H2O Scala API.
```
Comment (michalkurka) [inline h2o-docs/src/product/api-changes.rst:9]: > as the ``h2o-scala`` module was intended for internal use only and does not represent the H2O Scala API

This is not correct, the module was actually intended for public use, we even made examples how to use it independent of Sparkling Water: https://github.com/h2oai/h2o-droplets/blob/master/h2o-scala-droplet/

Specifically 
https://github.com/h2oai/h2o-droplets/blob/master/h2o-scala-droplet/build.gradle#L84

Comment (Pscheidl) [inline h2o-docs/src/product/api-changes.rst:9]:
```suggestion
The ``h2o-scala`` module is deprecated in version 3.30.0.5 and will be completely removed in the next major version, 3.32.0.1. This by no means has any impact on Scala support.
```

**Comment by Pscheidl** [comment]
I suggest to keep it simple @michalkurka. Removed the comment about origina intentions of `h2o-scala` module.

**Comment by michalkurka** [comment]
> This by no means has any impact on Scala support.

@Pscheidl I am all for keeping it simple, at the same time I do not understand how removing h2o-scala has no impact on Scala support.

File: h2o-docs/src/product/api-changes.rst [hunk 3]
```diff
@@ -3,6 +3,12 @@ API-Related Changes
 
 H2O-3 does its best to keep backwards compatibility between major versions, but sometimes breaking changes are needed in order to improve code quality and to address issues. This section provides a list of current breaking changes between specific releases.
 
+From 3.30.0.5
+~~~~~~~~~~~~~
+
+The ``h2o-scala`` module is deprecated in version 3.30.0.5 and will be completely removed in the next major version, 3.32.0.1. This by no means has any impact on Scala support.
```
Comment (Pscheidl) [inline h2o-docs/src/product/api-changes.rst:9]:
```suggestion
The ``h2o-scala`` module is deprecated in version 3.30.0.5 and will be completely removed in the next major version, 3.32.0.1.
```

**Comment by Pscheidl** [comment]
Another suggestion, let's keep it even more simple @michalkurka . WDYT ?

**Comment by michalkurka** [comment]
@angela0xdata should we remove the scaladoc links from the download page? http://h2o-release.s3.amazonaws.com/h2o/rel-zahradnik/4/index.html

**Comment by michalkurka** [comment]
+ there are other places in documentation we need to clean-up

**Pull request merged**